quodlibet/quodlibet/qltk/playorder.py

```diff
@@ -173,248 +230,345 @@ def next(self, playlist, iter):
+            plugin_cls.name = plugin.name
+        if plugin_cls.display_name is None:
+            plugin_cls.display_name = str(plugin.name).capitalize()
+        if plugin_cls.accelerated_name is None:
+            plugin_cls.accelerated_name = plugin_cls.display_name
+        self.append(plugin_cls)
 
+    def plugin_disable(self, plugin):
+        self.remove(plugin.cls)
 
-class PlayOrder(Gtk.Box, PluginHandler):
-    """A play order selection widget.
+DEFAULT_SHUFFLE_ORDERS = [OrderShuffle, OrderWeighted]
+DEFAULT_REPEAT_ORDERS = [RepeatSongForever, RepeatListForever]
 
-    Whenever something changes the 'changed' signal gets emitted.
 
-    TODO: split up in UI and management part
+class ToggledPlayOrderMenu(Gtk.Box):
+    """A toggle button with a menu button.
+    Items displayed are all `PlayOrder`
+
+    When the button is toggled, a `toggled` signal gets emitted.
     """
 
     __gsignals__ = {
-        'changed': (GObject.SignalFlags.RUN_LAST, None, ()),
+        'toggled': (GObject.SignalFlags.RUN_LAST, None, ()),
+        'changed': (GObject.SignalFlags.RUN_LAST, None, (object,)),
     }
 
-    def __init__(self, model, player):
-        super(PlayOrder, self).__init__()
-
-        self._model = model
-        self._player = player
-        self._plugins = []
-        self._inhibit_save = False
+    def __init__(self, icon_name, orders, current_order, enabled=False,
+                 tooltip=None, arrow_down=False):
+        """arrow_down -- the direction of the menu and arrow icon"""
+        print_d("Building new Toggler set to %r" % enabled)
+        assert issubclass(current_order, Order)
+        if current_order not in orders:
+            raise ValueError("%s is not supported by %s"
+                             % (current_order.__name__, orders))
 
-        self._shuffle = shuffle = ShuffleButton()
-        self.pack_start(shuffle, True, True, 0)
+        super(ToggledPlayOrderMenu, self).__init__()
+        self.__inhibit = True
 
-        if PluginManager.instance:
-            PluginManager.instance.register_handler(self)
+        context = self.get_style_context()
+        context.add_class(Gtk.STYLE_CLASS_LINKED)
 
-        self._set_order(
-            self._get_order(config.getboolean("memory", "shuffle")))
-        shuffle.connect("toggled", self._random_toggle)
+        self._toggle_button = toggle = Gtk.ToggleButton(
+            image=SymbolicIconImage(icon_name, Gtk.IconSize.SMALL_TOOLBAR))
+        if tooltip:
+            toggle.set_tooltip_text(tooltip)
+        toggle.set_active(enabled)
+        toggle.show_all()
+        qltk.remove_padding(toggle)
+        toggle.set_size_request(26, 26)
+        self.pack_start(toggle, True, True, 0)
 
-    def set_shuffle(self, value):
-        """Set shuffle, will change the order accordingly"""
+        def forward_signal(*args):
+            if self.__inhibit:
+                print_d("NOT forwarding programmatic toggled signal...")
```

Comment:
Did you want something like: https://lazka.github.io/pgi-docs/#GObject-2.0/functions.html#GObject.signal_handler_block ?